sdk/expo/src/index.ts

```diff
@@ -0,0 +1,49 @@
+import { ConfigPlugin, createRunOncePlugin } from '@expo/config-plugins';
+import withMoEngageAndroid from './android/withMoEngageAndroid';
+import { withMoEngageIos } from './apple';
+import { MoEngagePluginProps } from './types';
+
+const defaultProps: MoEngagePluginProps = {
+  android: {
+    disableMoEngageDefaultBackupFile: false,
+    configFilePath: 'assets/moengage/android_initilisation_config.xml',
+  },
+  apple: {
+    configFilePath: 'assets/moengage/MoEngage-Config.plist',
+    pushNotificationImpressionTrackingEnabled: true,
+    richPushNotificationEnabled: false,
+    pushTemplatesEnabled: false,
+    deviceTriggerEnabled: false,
+    liveActivityTargetPath: '',
+  },
+};
+
+/**
+ * Configure MoEngage SDK in your Expo app
+ */
+const withMoEngage: ConfigPlugin<MoEngagePluginProps | undefined> = (config, props) => {
+  // Apply default values or handle undefined props
+  const pluginProps: MoEngagePluginProps = props ? {
+    ...defaultProps,
```

Comment:
`android` and `apple` blocks seem redundant here. Please check on this.

Reply:
required to merge the nested object as it doesn't copy nested keys